src/hotspot/share/gc/z/zDirector.cpp

```diff
@@ -711,7 +711,12 @@ static ZWorkerCounts select_worker_threads(const ZDirectorStats& stats, uint you
     return {active_young_workers, active_old_workers};
   }
 
-  const double young_to_old_ratio = calculate_young_to_old_worker_ratio(stats);
+  double young_to_old_ratio = calculate_young_to_old_worker_ratio(stats);
```

Comment:
I think we might as well just clamp the return value in `calculate_young_to_old_worker_ratio`. Especially since it is a local helper function which is only used here. 
```c++
  return MIN2(old_vs_young_efficiency_ratio, (double)ZOldGCThreads);
```
It is suppose to calculate the worker ratio. And the largest ratio we can have is ZOldGCThreads old workers to 1 young worker.

Reply:
The ratio should probably also be called `old_to_young` as it is what it actually is, not `young_to_old`. But that is a preexisting issue which we can fix in a separate rfe.